### ISSUE TITLE
Issue with non-serializable classes in exceptions with fork VM

### ISSUE THREAD

**Issue by loicmathieu** [quick-perf/quickperf#75]
**Describe the bug**
When the VM is forked, all the test results from the forked VM are serialized to disk to be sent to the origin VM.

This uses Java Serialization.

If some non-serializable class is used inside the exception code of an exception thrown by the test, this leads to a crash of QuickPerf due to `java.lang.IllegalStateException`.

**Expected behavior**
(Describe the expected behavior clearly and concisely.)

**Actual behavior**

An error on Micronaut 2.0.0.M3 will generate the following stacktraces as it uses a non-serializable class

```
Unable to run test in a new JMV: Exception in thread "main" java.lang.IllegalStateException: Unable to save failures.
	at org.quickperf.repository.ObjectFileRepository.buildSerializationException(ObjectFileRepository.java:49)
	at org.quickperf.repository.ObjectFileRepository.save(ObjectFileRepository.java:44)
	at org.quickperf.repository.BusinessOrTechnicalIssueRepository.save(BusinessOrTechnicalIssueRepository.java:35)
	at org.quickperf.testlauncher.TestRunnerFromMain.executeTestMethod(TestRunnerFromMain.java:40)
	at org.quickperf.junit5.QuickPerfJunit5Core.main(QuickPerfJunit5Core.java:24)
Caused by: java.io.NotSerializableException: io.micronaut.http.client.exceptions.HttpClientErrorDecoder$1
	at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1185)
	at java.base/java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1553)
	at java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1510)
	at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1433)
	at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1179)
	at java.base/java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1553)
	at java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1510)
	at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1433)
	at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1179)
	at java.base/java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1553)
	at java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1510)
	at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1433)
	at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1179)
	at java.base/java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:349)
	at org.quickperf.repository.ObjectFileRepository.save(ObjectFileRepository.java:41)
	... 3 more

```

**Versions**
- QuickPerf: master
- JDK: 11
- OS: Unbuntu 20.04
- Database (if SQL annotation bug): H2

**Comment by jeanbisutti** [comment]
Thanks @loicmathieu for this report.

A way to fix this would be to update `TestIssue `class:

```java
 private static TestIssue convertThrowablesIntoToBusinessOrTechnicalIssue(List<Throwable> throwables) {
        Throwable firstThrowable = throwables.get(0);

        for (int i = 2; i < throwables.size(); i++) {
            Throwable throwable = throwables.get(i);
            firstThrowable.addSuppressed(throwable);
        }

        return new TestIssue(firstThrowable);
    }
```
In case where `firstThrowable` is not an instance of `Serializable`, then we may convert it into a Serializable `TestException` for example.

**Comment by jeanbisutti** [comment]
@loicmathieu Thanks for this report.

Could you please provide a test case?

**Comment by loicmathieu** [comment]
You can reproduce the issue by using the (under developement) micronaut-data example available here: https://github.com/loicmathieu/quickperf-examples/tree/micronaut-data-jdbc

You can launch the test via `mvn clean test -Dmaven.test.skip=false`.

**Comment by jeanbisutti** [comment]
@loicmathieu 

I don't get the same stack as you:

```
java.lang.IllegalStateException: Unable to deserialize failures.

	at org.quickperf.repository.ObjectFileRepository.buildDeserializationException(ObjectFileRepository.java:62)
	at org.quickperf.repository.ObjectFileRepository.find(ObjectFileRepository.java:57)
	at org.quickperf.sql.repository.SqlFileRepository.retrieveExecutedQueriesFromFile(SqlFileRepository.java:66)
	at org.quickperf.sql.repository.SqlFileRepository.findExecutedQueries(SqlFileRepository.java:55)
	at org.quickperf.sql.PersistenceSqlRecorder.findRecord(PersistenceSqlRecorder.java:62)
	at org.quickperf.sql.PersistenceSqlRecorder.findRecord(PersistenceSqlRecorder.java:26)
	at org.quickperf.issue.PerfIssuesEvaluator.findPerfRecord(PerfIssuesEvaluator.java:68)
	at org.quickperf.issue.PerfIssuesEvaluator.buildPerfRecordByAnnotation(PerfIssuesEvaluator.java:58)
	at org.quickperf.issue.PerfIssuesEvaluator.evaluatePerfIssues(PerfIssuesEvaluator.java:37)
	at org.quickperf.junit5.QuickPerfTestExtension.interceptTestMethod(QuickPerfTestExtension.java:75)
	at org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:149)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:140)
	at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:84)
	at org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
	at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:104)
	at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:98)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$6(TestMethodTestDescriptor.java:205)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:201)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:137)
	at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:71)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:135)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1540)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1540)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
	at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
	at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
	at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
	at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
	at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:248)
	at org.junit.platform.launcher.core.DefaultLauncher.lambda$execute$5(DefaultLauncher.java:211)
	at org.junit.platform.launcher.core.DefaultLauncher.withInterceptedStreams(DefaultLauncher.java:226)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:199)
	at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:132)
	at com.intellij.junit5.JUnit5IdeaTestRunner.startRunnerWithArgs(JUnit5IdeaTestRunner.java:69)
	at com.intellij.rt.junit.IdeaTestRunner$Repeater.startRunnerWithArgs(IdeaTestRunner.java:33)
	at com.intellij.rt.junit.JUnitStarter.prepareStreamsAndStart(JUnitStarter.java:230)
	at com.intellij.rt.junit.JUnitStarter.main(JUnitStarter.java:58)
	Suppressed: java.lang.NullPointerException
		at java.base/java.io.ObjectOutputStream$BlockDataOutputStream.getUTFLength(ObjectOutputStream.java:2142)
		at java.base/java.io.ObjectOutputStream$BlockDataOutputStream.writeUTF(ObjectOutputStream.java:2013)
		at java.base/java.io.ObjectOutputStream.writeUTF(ObjectOutputStream.java:872)
		at org.quickperf.sql.SqlExecution.writeExecutionInfo(SqlExecution.java:100)
		at org.quickperf.sql.SqlExecution.writeExternal(SqlExecution.java:91)
		at java.base/java.io.ObjectOutputStream.writeExternalData(ObjectOutputStream.java:1460)
		at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1431)
		at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1179)
		at java.base/java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:349)
		at java.base/java.util.ArrayDeque.writeObject(ArrayDeque.java:1184)
		at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
		at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
		at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
		at java.base/java.lang.reflect.Method.invoke(Method.java:566)
		at java.base/java.io.ObjectStreamClass.invokeWriteObject(ObjectStreamClass.java:1130)
		at java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1497)
		at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1433)
		at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1179)
		at java.base/java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1553)
		at java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1510)
		at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1433)
		at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1179)
		at java.base/java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:349)
		at org.quickperf.repository.ObjectFileRepository.save(ObjectFileRepository.java:41)
		at org.quickperf.repository.ObjectFileRepository.save(ObjectFileRepository.java:35)
		at org.quickperf.sql.repository.SqlFileRepository.flush(SqlFileRepository.java:45)
		at org.quickperf.sql.PersistenceSqlRecorder.stopRecording(PersistenceSqlRecorder.java:48)
		at org.quickperf.perfrecording.PerformanceRecording.stop(PerformanceRecording.java:39)
		at org.quickperf.junit5.QuickPerfTestExtension.executeTestMethodInNewJvmAndRecordPerformance(QuickPerfTestExtension.java:97)
		at org.quickperf.junit5.QuickPerfTestExtension.interceptTestMethod(QuickPerfTestExtension.java:67)
		at org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
		at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
		at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
		at org.junit.jupiter.engine.extension.TimeoutExtension.intercept(TimeoutExtension.java:149)
		at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestableMethod(TimeoutExtension.java:140)
		at org.junit.jupiter.engine.extension.TimeoutExtension.interceptTestMethod(TimeoutExtension.java:84)
		at org.junit.jupiter.engine.execution.ExecutableInvoker$ReflectiveInterceptorCall.lambda$ofVoidMethod$0(ExecutableInvoker.java:115)
		at org.junit.jupiter.engine.execution.ExecutableInvoker.lambda$invoke$0(ExecutableInvoker.java:105)
		at org.junit.jupiter.engine.execution.InvocationInterceptorChain$InterceptedInvocation.proceed(InvocationInterceptorChain.java:106)
		at org.junit.jupiter.engine.execution.InvocationInterceptorChain.proceed(InvocationInterceptorChain.java:64)
		at org.junit.jupiter.engine.execution.InvocationInterceptorChain.chainAndInvoke(InvocationInterceptorChain.java:45)
		at org.junit.jupiter.engine.execution.InvocationInterceptorChain.invoke(InvocationInterceptorChain.java:37)
		at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:104)
		at org.junit.jupiter.engine.execution.ExecutableInvoker.invoke(ExecutableInvoker.java:98)
		at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.lambda$invokeTestMethod$6(TestMethodTestDescriptor.java:205)
		at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
		at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.invokeTestMethod(TestMethodTestDescriptor.java:201)
		at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:137)
		at org.junit.jupiter.engine.descriptor.TestMethodTestDescriptor.execute(TestMethodTestDescriptor.java:71)
		at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:135)
		at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
		at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
		at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
		at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
		at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
		at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
		at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
		at java.base/java.util.ArrayList.forEach(ArrayList.java:1540)
		at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
		at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
		at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
		at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
		at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
		at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
		at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
		at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
		at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
		at java.base/java.util.ArrayList.forEach(ArrayList.java:1540)
		at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.invokeAll(SameThreadHierarchicalTestExecutorService.java:38)
		at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$5(NodeTestTask.java:139)
		at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
		at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$7(NodeTestTask.java:125)
		at org.junit.platform.engine.support.hierarchical.Node.around(Node.java:135)
		at org.junit.platform.engine.support.hierarchical.NodeTestTask.lambda$executeRecursively$8(NodeTestTask.java:123)
		at org.junit.platform.engine.support.hierarchical.ThrowableCollector.execute(ThrowableCollector.java:73)
		at org.junit.platform.engine.support.hierarchical.NodeTestTask.executeRecursively(NodeTestTask.java:122)
		at org.junit.platform.engine.support.hierarchical.NodeTestTask.execute(NodeTestTask.java:80)
		at org.junit.platform.engine.support.hierarchical.SameThreadHierarchicalTestExecutorService.submit(SameThreadHierarchicalTestExecutorService.java:32)
		at org.junit.platform.engine.support.hierarchical.HierarchicalTestExecutor.execute(HierarchicalTestExecutor.java:57)
		at org.junit.platform.engine.support.hierarchical.HierarchicalTestEngine.execute(HierarchicalTestEngine.java:51)
		at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:248)
		at org.junit.platform.launcher.core.DefaultLauncher.lambda$execute$5(DefaultLauncher.java:211)
		at org.junit.platform.launcher.core.DefaultLauncher.withInterceptedStreams(DefaultLauncher.java:226)
		at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:199)
		at org.junit.platform.launcher.core.DefaultLauncher.execute(DefaultLauncher.java:132)
		at org.quickperf.junit5.JUnit5TestRunner.runTestMethod(JUnit5TestRunner.java:58)
		at org.quickperf.junit5.JUnit5TestRunner.executeTestMethod(JUnit5TestRunner.java:35)
		at org.quickperf.testlauncher.TestRunnerFromMain.executeTestMethod(TestRunnerFromMain.java:54)
		at org.quickperf.testlauncher.TestRunnerFromMain.executeTestMethod(TestRunnerFromMain.java:35)
		at org.quickperf.junit5.QuickPerfJunit5Core.main(QuickPerfJunit5Core.java:24)
Caused by: java.io.EOFException
	at java.base/java.io.DataInputStream.readUnsignedShort(DataInputStream.java:345)
	at java.base/java.io.ObjectInputStream$BlockDataInputStream.readUnsignedShort(ObjectInputStream.java:3272)
	at java.base/java.io.ObjectInputStream$BlockDataInputStream.readUTF(ObjectInputStream.java:3328)
	at java.base/java.io.ObjectInputStream.readUTF(ObjectInputStream.java:1153)
	at org.quickperf.sql.SqlExecution.readExecutionInfo(SqlExecution.java:170)
	at org.quickperf.sql.SqlExecution.readExternal(SqlExecution.java:163)
	at java.base/java.io.ObjectInputStream.readExternalData(ObjectInputStream.java:2136)
	at java.base/java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2085)
	at java.base/java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1594)
	at java.base/java.io.ObjectInputStream.readObject(ObjectInputStream.java:430)
	at java.base/java.util.ArrayDeque.readObject(ArrayDeque.java:1208)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at java.base/java.io.ObjectStreamClass.invokeReadObject(ObjectStreamClass.java:1160)
	at java.base/java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2216)
	at java.base/java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2087)
	at java.base/java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1594)
	at java.base/java.io.ObjectInputStream.defaultReadFields(ObjectInputStream.java:2355)
	at java.base/java.io.ObjectInputStream.readSerialData(ObjectInputStream.java:2249)
	at java.base/java.io.ObjectInputStream.readOrdinaryObject(ObjectInputStream.java:2087)
	at java.base/java.io.ObjectInputStream.readObject0(ObjectInputStream.java:1594)
	at java.base/java.io.ObjectInputStream.readObject(ObjectInputStream.java:430)
	at org.quickperf.repository.ObjectFileRepository.find(ObjectFileRepository.java:55)
```

The first problem explaining the other parts of the stack trace:
```
Suppressed: java.lang.NullPointerException
		at java.base/java.io.ObjectOutputStream$BlockDataOutputStream.getUTFLength(ObjectOutputStream.java:2142)
		at java.base/java.io.ObjectOutputStream$BlockDataOutputStream.writeUTF(ObjectOutputStream.java:2013)
		at java.base/java.io.ObjectOutputStream.writeUTF(ObjectOutputStream.java:872)
		at org.quickperf.sql.SqlExecution.writeExecutionInfo(SqlExecution.java:100)
		at org.quickperf.sql.SqlExecution.writeExternal(SqlExecution.java:91)
		at java.base/java.io.ObjectOutputStream.writeExternalData(ObjectOutputStream.java:1460)
		at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1431)
		at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1179)
		at java.base/java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:349)
		at java.base/java.util.ArrayDeque.writeObject(ArrayDeque.java:1184)
		at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
		at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
		at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
		at java.base/java.lang.reflect.Method.invoke(Method.java:566)
		at java.base/java.io.ObjectStreamClass.invokeWriteObject(ObjectStreamClass.java:1130)
		at java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1497)
		at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1433)
		at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1179)
		at java.base/java.io.ObjectOutputStream.defaultWriteFields(ObjectOutputStream.java:1553)
		at java.base/java.io.ObjectOutputStream.writeSerialData(ObjectOutputStream.java:1510)
		at java.base/java.io.ObjectOutputStream.writeOrdinaryObject(ObjectOutputStream.java:1433)
		at java.base/java.io.ObjectOutputStream.writeObject0(ObjectOutputStream.java:1179)
		at java.base/java.io.ObjectOutputStream.writeObject(ObjectOutputStream.java:349)
		at org.quickperf.repository.ObjectFileRepository.save(ObjectFileRepository.java:41)
		at org.quickperf.repository.ObjectFileRepository.save(ObjectFileRepository.java:35)
		at org.quickperf.sql.repository.SqlFileRepository.flush(SqlFileRepository.java:45)
		at org.quickperf.sql.PersistenceSqlRecorder.stopRecording(PersistenceSqlRecorder.java:48)
		at org.quickperf.perfrecording.PerformanceRecording.stop(PerformanceRecording.java:39)
```

In `SqlFileRepository`: 
```
 public void flush(WorkingFolder workingFolder) {
        if (!this.flushed) {
            SqlExecutions executedQueries = this.sqlMemoryRepository.findExecutedQueries(workingFolder);
            if (!executedQueries.isEmpty()) {
                this.objectFileRepository.save(workingFolder, "sql.ser", executedQueries);
            }

            this.flushed = true;
        }

    }
```

`executedQueries` might be null.

Something seems to be missing in `micronaut-data-jdbc` example project.

**Comment by loicmathieu** [comment]
Thanks for debugging this, I naively think that the issue on the reproducer is the same but it's not.
Anyway, the orgininal issue is real, it's the reproducer that is wrong.

Can you re-open the issue? I'll try to create another reproducer.

**Comment by loicmathieu** [comment]
By the way, if `executedQueries` can be null, maybe add a null check in case someone uses an anotation to check that no queries has been issued ?

**Comment by jeanbisutti** [comment]
> By the way, if executedQueries can be null, maybe add a null check in case someone uses an anotation to check that no queries has been issued ?

If no query is raised, `executedQueries` is not supposed to be null.

**Comment by loicmathieu** [comment]
I think it's a but as `executionInfo` might not be readable via `readUTF` at :  org.quickperf.sql.SqlExecution.writeExecutionInfo(SqlExecution.java:100)

```
Time:1, Success:True, Type:Prepared, Batch:False, QuerySize:1, BatchSize:0, Query:["INSERT INTO `team` (`name`,`id`) VALUES (?,?)"], Params:[(Manchester United,1)]

	Time:1, Success:True, Type:Prepared, Batch:False, QuerySize:1, BatchSize:0, Query:["INSERT INTO `team` (`name`,`id`) VALUES (?,?)"], Params:[(Atlético de Madrid',2)]

	Time:0, Success:True, Type:Prepared, Batch:False, QuerySize:1, BatchSize:0, Query:["INSERT INTO `player` (`first_name`,`last_name`,`team_id`,`id`) VALUES (?,?,?,?)"], Params:[(Paul,Pogba,1,1)]

	Time:0, Success:True, Type:Prepared, Batch:False, QuerySize:1, BatchSize:0, Query:["INSERT INTO `player` (`first_name`,`last_name`,`team_id`,`id`) VALUES (?,?,?,?)"], Params:[(Antoine,Griezmann,2,2)]

	Time:3, Success:True, Type:Prepared, Batch:False, QuerySize:1, BatchSize:0, Query:["SELECT player_.`id`,player_.`first_name`,player_.`last_name`,player_.`team_id` FROM `player` player_"], Params:[()]

	Time:0, Success:True, Type:Prepared, Batch:False, QuerySize:1, BatchSize:0, Query:["SELECT team_.`id`,team_.`name` FROM `team` team_ WHERE (team_.`id` = ?)"], Params:[(1)]

	Time:0, Success:True, Type:Prepared, Batch:False, QuerySize:1, BatchSize:0, Query:["SELECT team_.`id`,team_.`name` FROM `team` team_ WHERE (team_.`id` = ?)"], Params:[(2)]
```

**Comment by loicmathieu** [comment]
The NPE issue was because the driver didn't set the dataSourceName and connectionId inside the Connection info when creating the connection proxy which generates a NPE when serializing the ExecutionInfo.

I'll try to create a reproducer for the original serialization issue.

**Comment by loicmathieu** [comment]
To reproduce the original issue, you can use the newly created micronaut-data-jdbc example project.
Then you replace the body of the `org.quickperf.micronaut.micronauttest.controller.PlayerController.findAll()` with the following code.

```java
throw new HttpClientResponseException("Hello", HttpResponse.serverError());
```

After launching the test with `mvn clean test -Dmaven.test.skip=false -Dmaven.test.failure.ignore -Dtest=PlayerControllerTest` you will see the issue.

**Comment by jeanbisutti** [comment]
Fixed by 7f0b31366410112ba4a8b8b3cce22d07aad9705c
